src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CipherData.cs

```diff
@@ -31,7 +33,7 @@ private bool CacheValid
             }
         }
 
-        public CipherReference CipherReference
+        public CipherReference? CipherReference
```

Comment:
This doesn't look right.  The setter is explicitly throwing an exception if value is null.  Same for CipherValue below.

Reply:
Yes, I missed the fact that the constructors call either one of two setters, each of which guarantee non-null.

Reply:
Actually, there is a default constructor too. I could be wrong, but I think these **do** need to be nullable...

Reply:
Making them nullable implies you can set them to null, but the setter explicitly throws for null.  If it's going to remain `public CipherReference? CipherReference`, then it also needs `[DisallowNull]`.

Reply:
But somebody _could_ do `new CipherData().CipherReference.ToString()` and get an NRE.  That would be pretty silly though, so could you clarify if we should be annotating for every concievable usage, or just for 'reasonable' usage. (or perhaps I've missed something completely...?)

Reply:
We're talking past each other. It should be:
```C#
[DisallowNull]
public CipherReference? CipherReference
```
which says the property may return null but can't be set to null.

Reply:
I see. Thank you.